oci/layer/utils.go

```diff
@@ -229,11 +230,19 @@ func InnerErrno(err error) error {
 
 // isOverlayWhiteout returns true if the FileInfo represents an overlayfs style
 // whiteout (i.e. mknod c 0 0) and false otherwise.
-func isOverlayWhiteout(info os.FileInfo) bool {
-	sysStat := info.Sys().(*unix.Stat_t)
-	if unix.Major(uint64(sysStat.Rdev)) != 0 || unix.Minor(uint64(sysStat.Rdev)) != 0 {
-		return false
+func isOverlayWhiteout(info os.FileInfo) (bool, error) {
+	var major, minor uint32
+	switch stat := info.Sys().(type) {
+	case *unix.Stat_t:
+		major = unix.Major(uint64(stat.Rdev))
+		minor = unix.Minor(uint64(stat.Rdev))
+	case *syscall.Stat_t:
+		major = unix.Major(uint64(stat.Rdev))
+		minor = unix.Minor(uint64(stat.Rdev))
+	default:
+		return false, errors.Errorf("unknown stat info type %T", info.Sys())
```

Comment:
Alright, we can agree to disagree on this (and since `umoci` is used as a library I do get your point). But I would like you add something like `[internal error]` or `[programmer error]` to the start of the error message to make sure it's clear that this is a programming error and not some other type of error.